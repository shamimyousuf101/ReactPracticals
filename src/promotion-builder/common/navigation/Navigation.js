import React from "react";
import PropsTypes from "prop-types";
import bem from "bem-cn";

import "./navigation.css";
import NavLinks from "./NavLinks"; 
import { VIEW , NAVIGATION_DATA } from "../../../constants/constants";


const b = bem("navigation");

const Navigation = ({ updateView, resetPrommotionId }) => {
  
  const onMenuClick = navId => {
    switch (navId) {
      case "SEARCH":
        updateView(VIEW.SEARCH);
        return;
      case "ASSET":
        updateView(VIEW.ASSET_MANAGER);
        return;
      case "EDIT":
        resetPrommotionId();
        updateView(VIEW.SAVE_PROMOTION);
        return;
      default:
        updateView(VIEW.SEARCH);
    }
  };

  return (
    <nav className={b()}>
      <ul>
        {NAVIGATION_DATA.map(({ navId, label }) => (
          <NavLinks
            label={label}
            navId={navId}
            clickHandler={() => onMenuClick(navId)}
          />
        ))}
      </ul>
    </nav>
  );
};

Navigation.propTypes = {
  clickHandler: PropsTypes.func.isRequired
};

export default Navigation;
