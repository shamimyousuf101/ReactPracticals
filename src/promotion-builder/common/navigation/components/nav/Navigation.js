import React from 'react';

import  Links from '../links/Links';
import { navigationData } from '../../navigationData'

import './navigation.css';

const Navigation = ({changeView}) => {  


    return (
    <nav className="menu">
        <ul>
            {
            navigationData.map( ({className, view, text}) => (<Links href={'#'} className={className} key={text} text={text} menuClickHandler={() => changeView(view)}/>) )
            }
        </ul>
    </nav>
    )
}

export default Navigation;



