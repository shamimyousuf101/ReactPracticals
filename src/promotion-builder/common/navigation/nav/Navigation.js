import React from 'react';
import PropsTypes from 'prop-types';

import Links from '../links/Links';

import './navigation.css';

const Navigation = ({menuData, clickHandler}) => {  
    return (
    <nav className="menu">
        <ul>
            {
            menuData.map( ({className, text}) => (<Links href={'#'} className={className} key={text} text={text} clickHandler={clickHandler}/>) )
            }
        </ul>
    </nav>
    )
}

Navigation.propTypes = {
    menuData: PropsTypes.array.isRequired,
    clickHandler: PropsTypes.func.isRequired
}

export default Navigation;