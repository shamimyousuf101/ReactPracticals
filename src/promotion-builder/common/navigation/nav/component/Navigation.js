import React from 'react';

import Links from '../../links/Links';
import { navigationData } from '../../navigationData'

import './navigation.css';

const Navigation = () => {  
    return (
    <nav className="menu">
        <ul>
            {
            navigationData.map( ({className, text}) => (<Links href={'#'} className={className} key={text} text={text}/>) )
            }
        </ul>
    </nav>
    )
}

export default Navigation;



