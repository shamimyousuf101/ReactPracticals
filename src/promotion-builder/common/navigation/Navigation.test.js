import React from "react";
import renderer from 'react-test-renderer';

import Navigation from "./Navigation";


describe("Navigation component", () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Navigation
        updateView={jest.fn()}
        clearSearchTerm={jest.fn()}
        resetPrommotionId={jest.fn()}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
