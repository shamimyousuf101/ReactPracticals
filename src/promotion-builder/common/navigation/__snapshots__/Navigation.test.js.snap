// Jest Snapshot v1, https://goo.gl/fbAQLP

//TODO - somehting is wrong

exports[`Navigation component renders correctly 1`] = `
<nav
  className="navigation"
>
  <ul>
    <li
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
      onClick={[Function]}
    >
      <a
        href="#"
      >
        Search
      </a>
    </li>
    <li
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
      onClick={[Function]}
    >
      <a
        href="#"
      >
        Asset Manager
      </a>
    </li>
    <li
      className={
        Object {
          "has": [Function],
          "is": [Function],
          "mix": [Function],
          "split": [Function],
          "state": [Function],
          "toString": [Function],
        }
      }
      onClick={[Function]}
    >
      <a
        href="#"
      >
        Create / Edit
      </a>
    </li>
  </ul>
</nav>
`;
